apps/web/src/components/ProfileDocument/ProfileDocument.tsx

```diff
@@ -162,17 +166,37 @@ const ProfileDocument = React.forwardRef<HTMLDivElement, ProfileDocumentProps>(
                   <React.Fragment key={result.id}>
                     <div>
                       <PageSection>
-                        <Heading level="h2">
-                          {(!!result.firstName || !!result.lastName) &&
-                            getFullNameLabel(
-                              result.firstName,
-                              result.lastName,
-                              intl,
-                            )}
+                        <Heading level="h2" data-h2-font-weight="base(700)">
+                          {anonymous ? (
+                            <>
+                              {getFullNameLabel(
+                                result.firstName,
+                                `${result.lastName?.slice(0, 1)}.`,
```

Comment:
I beleive this will result in printing "John Undefined" if they have no last name set.

REF: [TypeScript Playground, undefined slice](https://www.typescriptlang.org/play?#code/C4TwDgpgBAqgzhATlAvFA3gKAJADMCWicwAcgIYC2EA-AFxTGL4B2A5lAD5TMCuANn05QezACYQCzCKIDcOPmWLkqdBsCZshvAUJHjJ0uQF9MmAMYB7ZsWEIi9eEgDaAXVRQnOLNjyEllCHoAIgApMikggBocbAV-KmCAEQsIIJwjaOxvXyJSAOCQiwALZjTsExc5cysbcKsQCnwAL2l3Hjs4ADoKMjAACnakVAA+KD7vAlzlQNskTsn4iEy4vISoAAMAEnRBxE6V6epOuD58Mwg+gAZIqABGAEojdcwje-vTS2sLPgh9i1Y+nVmA1mtJ7jIgA)

Reply:
Nice catch thanks for that ref too ❤️ Fixed by [6b0a0f0](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7747/commits/6b0a0f04d8846c7a5c123b139cedaa4d53a3df79)